src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -193,19 +194,26 @@ else if (FlutterSettings.getInstance().isDisablePreviewDart2()) {
       args.add("--no-preview-dart-2");
     }
 
-    if (!FlutterSettings.getInstance().isDisablePreviewDart2() && FlutterSettings.getInstance().isTrackWidgetCreation()) {
-      args.add("--track-widget-creation");
+    if (flutterLaunchMode == FlutterLaunchMode.DEBUG) {
+      if (!FlutterSettings.getInstance().isDisablePreviewDart2() && FlutterSettings.getInstance().isTrackWidgetCreation()) {
+        args.add("--track-widget-creation");
```

Comment:
we might actually want to put this fix into the release. Shipping release builds with track widget location was bad.

Reply:
I ran across this when thinking about how this might interact with launching apps in `--profile` mode; it would be a good fix for the milestone.